### PR TITLE
Update deps to latest version of core and identity

### LINKED ISSUE #3533: Update deps to latest azure core / identity versions

### PR DESCRIPTION
**Pull request by gearama**
closes #3533

Updating to the latest version of dependancies for keyvault sdks 

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-certificates/CMakeLists.txt [hunk 1]
```diff
@@ -40,9 +40,11 @@ if(FETCH_SOURCE_DEPS)
         add_subdirectory(${oneFolder} EXCLUDE_FROM_ALL)
     endforeach()
 elseif(NOT AZ_ALL_LIBRARIES)
-  find_package(azure-core-cpp "1.2.0" CONFIG QUIET)
+  find_package(azure-core-cpp "1.5.0" CONFIG QUIET)
+  find_package(azure-identity-cpp "1.2.0" CONFIG QUIET)
   if(NOT azure-core-cpp_FOUND)
```
Comment (LarryOsterman) [inline sdk/keyvault/azure-security-keyvault-certificates/CMakeLists.txt:45]: Why the new dependency on identity? Was the dependency already there but hidden? Why didn't the missing dependency make a difference?

File: cmake-modules/FolderList.cmake [hunk 2]
```diff
@@ -2,16 +2,16 @@ macro(GetFolderList project)
...
     message ("FLAG VALUE : ${FETCH_SOURCE_DEPS}")
     if(${project} STREQUAL CERTIFICATES)
-        DownloadDepVersion(sdk/core azure-core 1.2.0)
-        DownloadDepVersion(sdk/identity azure-identity 1.1.0)
+        DownloadDepVersion(sdk/core azure-core 1.5.0)
+        DownloadDepVersion(sdk/identity azure-identity 1.2.0)
     elseif(${project} STREQUAL IDENTITY)
         DownloadDepVersion(sdk/core azure-core 1.2.0)
     elseif(${project} STREQUAL SECRETS)
-        DownloadDepVersion(sdk/core azure-core 1.2.0)
```
Comment (LarryOsterman) [inline cmake-modules/FolderList.cmake:10]: What's the purpose of this file? Is this something that should be updated for all SDKs? If so, should Template and Attestation be listed?

File: sdk/keyvault/azure-security-keyvault-certificates/vcpkg/Config.cmake.in [hunk 3]
```diff
@@ -4,7 +4,8 @@
 @PACKAGE_INIT@
 
 include(CMakeFindDependencyMacro)
-find_dependency(azure-core-cpp "1.3.1")
+find_dependency(azure-core-cpp "1.5.0")
+find_dependency(azure-identity-cpp "1.2.0")
```
Comment (antkmsft) [inline sdk/keyvault/azure-security-keyvault-certificates/vcpkg/Config.cmake.in:8]: No client should depend on identity. Core provides all interfaces, Identity provides implementations for them. It is the user app that takes dependency on un uses service client library and identity library, but not that the client library uses identity directly, or depends on it.

**Review by antkmsft — CHANGES_REQUESTED** [review]

File: cmake-modules/FolderList.cmake (hunk 2 above)
Comment (vhvb1989) [reply cmake-modules/FolderList.cmake:10]: This file provides the foundation for `FETCH_SOURCE_DEPS` option.  People can use that option when building one only package and cmake would fetch the AzureSDK dependencies using this file configuration

**Review by vhvb1989 — COMMENTED** [review]
+1 to Anton's comment.

File: cmake-modules/FolderList.cmake (hunk 2 above)
Comment (LarryOsterman) [reply cmake-modules/FolderList.cmake:10]: Should this be updated for all SDKs? Or just the set listed?

File: sdk/keyvault/azure-security-keyvault-certificates/vcpkg/Config.cmake.in (hunk 3 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-certificates/vcpkg/Config.cmake.in:8]: sounds good  , thank you for the clarification

File: cmake-modules/FolderList.cmake (hunk 2 above)
Comment (gearama) [reply cmake-modules/FolderList.cmake:10]: we can add other SDKs, as wanted

**Review by antkmsft — APPROVED** [review]

File: cmake-modules/FolderList.cmake [hunk 4]
```diff
@@ -2,16 +2,13 @@ macro(GetFolderList project)
...
+        DownloadDepVersion(sdk/core azure-core 1.5.0)
     elseif(${project} STREQUAL IDENTITY)
         DownloadDepVersion(sdk/core azure-core 1.2.0)
     elseif(${project} STREQUAL SECRETS)
-        DownloadDepVersion(sdk/core azure-core 1.2.0)
-        DownloadDepVersion(sdk/identity azure-identity 1.1.0)
+        DownloadDepVersion(sdk/core azure-core 1.5.0)
     elseif(${project} STREQUAL KEYS)
-        DownloadDepVersion(sdk/core azure-core 1.2.0)
-        DownloadDepVersion(sdk/identity azure-identity 1.1.0)
```
Comment (antkmsft) [inline cmake-modules/FolderList.cmake:14]: I don't think identity should be removed from here

File: eng/vcpkg-commit.txt [hunk 5]
```diff
@@ -1 +1 @@
-be5c4ef68b51142ba705f0678b45d284977de677
+c556911a79f8ab23b12648ddfe8a5444a6c05080
```
Comment (antkmsft) [inline eng/vcpkg-commit.txt:1]: Unintended change?

File: cmake-modules/FolderList.cmake (hunk 4 above)
Comment (gearama) [reply cmake-modules/FolderList.cmake:14]: true

File: eng/vcpkg-commit.txt (hunk 5 above)
Comment (gearama) [reply eng/vcpkg-commit.txt:1]: no, we have this amazing thing in the cmakegenerationbuild steps  that checks out vcpkg to a specific commit , that is the SHA of the commit to which vcpkg will sync. the previous version was before the release of core 1.5, the second is after 1.5 was released.

Comment (LarryOsterman) [reply eng/vcpkg-commit.txt]: FWIW, I'm hoping to get rid of this entirely, with @danieljurek 's help.

Comment (LarryOsterman) [reply eng/vcpkg-commit.txt]: In my PR, this entire thing is redundant, instead we use the baseline-commit feature of the vcpkg.json file to handle this.

**Pull request merged**